### PR TITLE
AWS deployment

### PR DESCRIPTION
**Pull request by jancona**
Using Aurora Serverless Postgres, SQS and Elasticsearch.

### CODE REVIEW

File: db/db_setup.sh [hunk 1]
```diff
@@ -1,4 +1,4 @@
 #!/bin/bash
 port=${1:-5432} # can override port from command line
 PGPASSWORD=postgres psql -U postgres -h localhost -p $port -f create_db.sql
-migrate -database "postgres://postgres:postgres@localhost:$port/cms?sslmode=disable" -path migrations up
+migrate -database "postgres://ourroots_schema:password@localhost:$port/cms?sslmode=disable" -path migrations up
```
Comment (DallanQ) [inline db/db_setup.sh:4]: just curious, why a separate username for the migration?

File: db/migrations/000004_new_tables.up.sql [hunk 2]
```diff
@@ -5,43 +5,8 @@ CREATE TABLE cms_user (
   insert_time TIMESTAMP WITH TIME ZONE DEFAULT CURRENT_TIMESTAMP,
   last_update_time TIMESTAMP WITH TIME ZONE DEFAULT CURRENT_TIMESTAMP
 );
--- TODO: Is tit worth using a more specific index, e.g. on (Issuer,Subject)?
+-- TODO: Is it worth using a more specific index, e.g. on (Issuer,Subject)?
 CREATE INDEX idx_user_body ON cms_user USING GIN (body);
```
Comment (DallanQ) [inline db/migrations/000004_new_tables.up.sql:9]: I'm not sure we need a GIN index on the entire body.  If we need them on specific attributes, we can create them on those attributes: "using gin ((body->'attrName'))". But at least so far, I can't think of where we reference body attributes in WHERE clauses.

File: db/db_setup.sh (hunk 1 above)
Comment (jancona) [reply db/db_setup.sh:4]: So the normal app user doesn't deed permission to drop tables. `ourroots_schema` is set up as the equivalent of the `postgres` user in a normal install.

File: db/migrations/000004_new_tables.up.sql (hunk 2 above)
Comment (jancona) [reply db/migrations/000004_new_tables.up.sql:9]: I'm fine either way. This shouldn't be a big table so it probably doesn't matter much.

File: uglyui/src/services/Server.js [hunk 3]
```diff
@@ -2,7 +2,8 @@ import axios from "axios";
 import { getInstance } from "../auth";
 
 const apiClient = axios.create({
-  baseURL: `http://localhost:8000`,
+  // baseURL: `http://localhost:8000`,
+  baseURL: `/api`,
```
Comment (DallanQ) [inline uglyui/src/services/Server.js:6]: How does this work? Doesn't the server run on a different port than the client when you run locally?

Comment (jancona) [reply uglyui/src/services/Server.js:6]: Good catch. Can we put that in a config file that I can replace when uploading to AWS?

File: db/db_setup.sh (hunk 1 above)
Comment (DallanQ) [reply db/db_setup.sh:4]: ok, that makes sense.

File: uglyui/src/services/Server.js (hunk 3 above)
Comment (DallanQ) [reply uglyui/src/services/Server.js:6]: What I've done in the past is pretty ugly: as part of the production build process I run "sed" on a constants file. I'll see if webpack has improved that situation and let you know.

**Pull request merged**